core/src/test/java/org/elasticsearch/recovery/RelocationIT.java

```diff
@@ -424,22 +424,23 @@ public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) throws IO
 
     public void testIndexAndRelocateConcurrently() throws ExecutionException, InterruptedException {
         int halfNodes = randomIntBetween(1, 3);
-        Settings blueSetting = Settings.builder().put("node.attr.color", "blue").build();
-        InternalTestCluster.Async<List<String>> blueFuture = internalCluster().startNodesAsync(halfNodes, blueSetting);
-        Settings redSetting = Settings.builder().put("node.attr.color", "red").build();
-        InternalTestCluster.Async<java.util.List<String>> redFuture = internalCluster().startNodesAsync(halfNodes, redSetting);
-        blueFuture.get();
-        redFuture.get();
-        logger.info("blue nodes: {}", blueFuture.get());
-        logger.info("red nodes: {}", redFuture.get());
+        Settings[] nodeSettings = Stream.concat(
+            Stream.generate(() -> Settings.builder().put("node.attr.color", "blue").build()).limit(halfNodes),
+            Stream.generate(() -> Settings.builder().put("node.attr.color", "red").build()).limit(halfNodes)
+            ).toArray(Settings[]::new);
```

Comment:
it was clearer but the problem with the old way is that the test cluster had no idea how many nodes are coming.  That was annoying as you can not go directly from a 2 nodes cluster to a 4 nodes cluster (a 2 nodes cluster won't let you update min master nodes to 3) so you'll need to start those nodes one by one. I found the current "start 4 nodes with a bang" simpler compared to that